src/go/rpk/pkg/cli/cmd/wasm/list.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package wasm
+
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+
+	"github.com/Shopify/sarama"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/cli/ui"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/kafka"
+)
+
+type coprocInfo struct {
+	InputTopics []string `json:"input_topics"`
+	Description string   `json:"description"`
+}
+
+type coprocNodeInfo struct {
+	NodeId       int                   `json:"node_id"`
+	Status       string                `json:"status"`
+	Coprocessors map[string]coprocInfo `json:"coprocessors"`
+}
+
+func NewListCommand(
+	clientClosure func() (sarama.Client, error),
+	consumerClosure func(sarama.Client) (sarama.Consumer, error),
+) *cobra.Command {
+	var pretty bool
+
+	command := &cobra.Command{
+		Use:   "list",
+		Short: "display the current status of functions within redpanda",
+		RunE: func(_ *cobra.Command, args []string) error {
+			return wasmList(clientClosure, consumerClosure, pretty)
+		},
+	}
+
+	command.Flags().BoolVar(&pretty, "raw", false, "Print the raw json to stdout")
+	return command
+}
+
+func printResults(dataSet *map[int64]coprocNodeInfo, raw bool) {
+	if raw {
+		stringRep, _ := json.Marshal(dataSet)
+		log.Info(string(stringRep))
+		return
+	}
+	t := ui.NewRpkTable(log.StandardLogger().Out)
+	t.SetColWidth(80)
+	t.SetAutoWrapText(false)
+	header := [...]string{"Script Name", "Node ID", "Input Topics", "Description"}
```

Comment:
So just in this file, there are 3 terms to refer to the same thing: "coproc", "function", and "script". We should stick to one across the codebase, and if it's changed later, it will be easier to find and replace.